korgw-sample/src/commonMain/kotlin/com/soywiz/korgw/sample/KorgwSample.kt

```diff
@@ -68,144 +53,103 @@ class KCube {
         name = "MY_PROG"
     )
 
-    fun run() = DefaultGameWindow.mainLoop {
-            configure(600, 600, "Multicolored cube", fullscreen = false)
+    var projMat = Matrix3D().setToPerspective(aov, ar, near, far)
+    var viewMat = Matrix3D().setToTranslation(xPos, yPos, zPos)
+    var modlMat = Matrix3D().identity()
+
+    fun run() = Korio {
+        DefaultGameWindow.loop {
+            //TODO: why can't I use width and height vars here ?
+            configure(400, 400, "Multicolored cube", fullscreen = false)
+            addEventListener<ReshapeEvent> {
+                reshape(it, ag)
+            }
             addEventListener<RenderEvent> {
                 render(ag)
             }
         }
-
-    fun init(ag:AG) {
-        //val gl = (ag as AGAwt).gl ??
-        //gl.clearColor(0f,0f,1f,1f)
-        //gl.clearDepthf(1.0f)
-        //gl.enable(gl.DEPTH_TEST)
-        // gl.depthFunc(gl.LEQUAL)
     }
 
-    private var vertices: AG.Buffer? = null
-    private var verticesData: FloatArrayList? = null
+    fun reshape(event: ReshapeEvent, ag: AG) {
+        width = event.width
+        height = event.height
+        if (height <= 0)
+            height = 1
+        ar = (width.toFloat() / height.toFloat())
+        ag.setViewport(0, 0, width, height)
+        projMat = Matrix3D().setToPerspective(aov, ar, near, far)
```

Comment:
Since that method mutates the content: `projMat.setToPerspective(aov, ar, near, far)`
(you already have an instance created)

Reply:
ok